setup.py

```diff
@@ -95,7 +95,11 @@ def build_extension(self, ext):
                 cmake_args += ["-A", "x64"]
             build_args += ["--", "/m"]
         else:
-            build_args += ["--", "-j"]
+            try:
+                CPU_COUNT = os.cpu_count()  # python 3.4+
```

Comment:
Docs suggest `len(os.sched_getaffinity(0))` instead…

Reply:
cool; now I need to know why - I assume it interacts better w/ `cgroups`; wonder what it does on the mac

Reply:
Doc says `They are only available on some Unix platforms.  For more detailed information, consult your Unix manpages.`

I'll look more into it.

Reply:
Can we just remove the concurrency? It's not like `libsonata` is a very large library that absolutely needs a parallel build.

Reply:
We could remove concurrency, but it's pretty easy to have, and it does make install faster (13s vs. 17s on my old & slow machine)

I just wanted to fix it so it doesn't start *all* the compiles at the same time - `make -j` isn't very smart.

Reply:
Seems like not that much gain…

Reply:
30% for 3 lines of code?  Seems like a no effort win to me, but if it's a no go on your end, let me know and I'll remove it.

Reply:
Well, it's 30%, but also just 4 seconds. I guess the proper way would be to first test for `sched_getaffinity`, then for `cpu_count`, and then fall back to not having any parallelism? Also safeguard for the fact that any of these methods may just return 0…

Seems like this could be a feature of setuptools by now…

Reply:
When you're building the set of packages w/ `spack`, how does it manage parallelism? Does it build the packages serially? How many packages are you building? Because 4 seconds probably adds up over time if they are build serially.

Also, at some point this repo will have the reporting stuff in it, afaik, so I'm not sure it's worth removing the parallelism, to potentially put it back in when the compile time increases.

Finally, the main point of this change is to remove the bare `make -j`, which can launch way too many processes.  Whether I use exactly the number of CPUs, or a smaller subset, the behavior is better than saturating the scheduler, and better than compiling things serially, IMO.

Reply:
IIRC, spack calls `make`/`cmake` with some user-specified parallelism (36 for us). There are bigger axes to grind than pure build times in our setup, though.

Anyways, I'm not against building in parallel, but more lamenting that this is not better integrated in `setuptools` and the like. If we end up doing it like I outlined above, we should probably comb through our repositories and fix other instances (i.e., MVDTool).